### PR TITLE
igUploader send multiple files in one request feature

### LINKED ISSUE #1501: Implement multiple files upload in a single XHR
Implement uploading multiple files in a single XML HTTP Request.

### Expected result  
An option to allow sending multiple files in a single XHR. Similar to what is available in JavaScript:

function uploadFiles(files) {
    var xhr = new XMLHttpRequest();
    var formData = new FormData();
    xhr.onload = successfullyUploaded;
    xhr.open("POST", "http://localhost:3000/upload", true);
    xhr.setRequestHeader('X-Requested-With','XMLHttpRequest');
    for(var file in files) {
        formData.append("uploads", files[file].data);
    }
    xhr.send(formData);
}

### PR DESCRIPTION
**Pull request by DDavidkov**
Closes #1501

Implement the igUploader "Multiple files upload in a single XHR"

### CODE REVIEW

File: src/js/modules/infragistics.ui.upload.js [hunk 1]
```diff
@@ -1545,7 +1545,22 @@
...
+				});
+
+				//Get
+				var sendDataInOneRequest = $(".selector").igUpload("option", "sendDataInOneRequest");
+
+				//Set
+				$(".selector").igUpload("option", "sendDataInOneRequest", true);
+			```
+			*/
+			sendDataInOneRequest: false
```
Comment (kdinev) [inline src/js/modules/infragistics.ui.upload.js]: Maybe we should name it `singleRequest` or `useSingleRequest`? @SlavUI What do you think?

File: src/js/modules/infragistics.ui.upload.js [hunk 2]
```diff
@@ -2355,32 +2370,121 @@
...
+						key += currentKey;
+					} else {
+						key += ";" + currentKey;
+					}
+
+					fileId.push(currentFileId);
+
+					self._trigger(self.events.onFormDataSubmit,
+						null,
+						{ formData: formData,
```
Comment (kdinev) [inline src/js/modules/infragistics.ui.upload.js]: Formatting.

File: src/js/modules/infragistics.ui.upload.js (hunk 1 above)
Comment (mpavlinov) [reply src/js/modules/infragistics.ui.upload.js]: I'm for "singleRequest".

Comment (gedinakova) [reply src/js/modules/infragistics.ui.upload.js]: I vote for useSingleRequest as it implies what value is expected.

Comment (SlavUI) [reply src/js/modules/infragistics.ui.upload.js]: Let's call it useSingleRequest :)

**Comment by kdinev** [comment]
@DDavidkov Convert the spaces into tabs for the whole file please.

**Comment by gedinakova** [comment]
Related #1532

**Pull request merged**